content/departments/engineering/dev/tools/grpc_cs_enablement.md

```diff
@@ -0,0 +1,165 @@
+
+```yaml
+- name: zoekt-indexserver
+  env:
+    - name: GRPC_ENABLED
+      value: 'false'
+  image: docker.io/sourcegraph/search-indexer:5.2.0
+```
+
+_zoekt-indexserver can’t read from Sourcegraph’s site configuration, so we can only use environment variables to communicate this setting._
+
+## How do you know if there might be a problem with gRPC?
+
+_See the gRPC [monitoring guide](./grpc.md) for more background._
+
+It’s important to understand that gRPC is merely the transport mechanism that our entire application logic is built on top of. Our existing monitoring, alerting, and reliance on user-submitted bug reports are our best tools for finding out if there might be a problem with the underlying application.
+
+However, we do have two tools that can help find some problems with gRPC.
+
+### gRPC Grafana dashboards
+
+We’ve included a standard set of dashboards for every gRPC service. (Example: sourcegraph.com’s [gitserver](https://sourcegraph.com/-/debug/grafana/d/gitserver/git-server?orgId=1)’s “GRPC Server Metrics” and “GRPC “Internal Error” Metrics” dashboards).
+
+These dashboards include request rates and error rates (along with the failing status codes) for every method.
+
+**However, note that these error rates dashboards don’t tell the whole story.**
+
+- These dashboards capture errors that occur for any reason (including application logic issues) that might be unrelated to GRPC itself.
+  - (For example, “cloning a repo that doesn’t exist” is a possible error that can be captured that doesn’t necessarily indicate a problem with grpc itself.)
+- Some failure modes are expected and not a cause for concern. For example, `status.DeadlineExceeded` or `status.Canceled` is frequently returned when the caller intentionally cancels an operation and can be entirely intentional (e.g., there is no need to continue streaming the output of a gitserver `archive` call if the user cancels a search request)
+- In many cases, we have never had an existing baseline “error rate” metrics for a given method. The error rate that you’re seeing might entirely normal for our application, it’s just being surfaced for the first time because of these new standardized dashboards.
+
+### Internal error reporter
+
+For a **subset of errors** that we are certain originate from the underlying gRPC libraries or configuration, we do have the “internal error” reporter (mentioned in the [monitoring guide](./grpc.md#more-on-internal-errors)) that can capture these.
+
+If you see either:
+
+- relevant entries in the “GRPC “internal errors” metrics**”** grafana dashboards
+- **OR**: application logs that are prefixed with `grpc.internal.error.reporter`
+
+That indicates a surefire issue with our gRPC implementation and (should be reported to us).
+
+## Who do you contact if there is a problem?
+
+Contact us in the #job-fair-grpc Slack channel.
```

Comment:
Should we maybe add `#team-apis` too? Just to make sure that this here outlives the channel in case we decide after Job Fair ends to rename/archive it.